### PR TITLE
chore: upload coverage of unittests together

### PR DESCRIPTION
**Pull request by matejvasek**
Upload results of unit tests on various operating systems at once as opposed to uploading them individually.

### CODE REVIEW

**Comment by matejvasek** [comment]
PTAL @dsimansk

File: .github/workflows/test-unit.yaml [hunk 1]
```diff
@@ -36,7 +36,23 @@ jobs:
           FUNC_REPO_BRANCH_REF: ${{ github.head_ref }}
       - name: Template Unit Tests
         run: make test-templates
+      - name: "Archive code coverage results"
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-${{ matrix.os }}
+          path: ./coverage.txt
+          retention-days: 7
```
Comment (dsimansk) [inline .github/workflows/test-unit.yaml:44]: It seems a bit long. I'd go with hours, e.g. 8h given it's per PR artifact and might iterate quickly.

Comment (matejvasek) [reply .github/workflows/test-unit.yaml:44]: Would 7 days hurt anything?

Comment (matejvasek) [reply .github/workflows/test-unit.yaml:44]: Also minimum is probably 1 day.

Comment (dsimansk) [reply .github/workflows/test-unit.yaml:44]: not sure who pays for storage :)

I'm not really, it's very small file. But 1 day is plenty.

Comment (matejvasek) [reply .github/workflows/test-unit.yaml]: >not sure who pays for storage :)

Uncle Bill?

**Review by dsimansk — APPROVED** [review]

**Comment by dsimansk** [comment]
@matejvasek wrt/ Codecov ~6% drop. Isn't unified uploader missing some reports?

**Comment by matejvasek** [comment]
> @matejvasek wrt/ Codecov ~6% drop. Isn't unified uploader missing some reports?
> 
> Update, actually it's stated in the dashboard. Hence different count of reports shouldn't be a problem, if we have same content, right?

Yes, there should be less reports since three reports were merged into one. I have no idea why the coverage decreased.

**Comment by dsimansk** [comment]
> > @matejvasek wrt/ Codecov ~6% drop. Isn't unified uploader missing some reports?
> > Update, actually it's stated in the dashboard. Hence different count of reports shouldn't be a problem, if we have same content, right?
> 
> Yes, there should be less reports since three reports were merged into one. I have no idea why the coverage decreased.

Strangely, it reports head 14 uploads vs. now 6 uploads. It's really off. 
Anyway, I'd suggest to keep monitoring the PR wrt/ codecov API rate limiting.  WDYT?

**Review by dsimansk — APPROVED** [review]

**Pull request merged**